File: Softeer/Lv3.스마트_물류.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
factory = list(input())

answer = 0
for i in range(len(factory)):
    if factory[i] == 'P':
        flag = 0
        for j in range(i-k, i+k+1):
            if j<0 or j==i or j>=len(factory):
                continue
            if factory[j] == 'H':
                answer += 1
                factory[j] = 'N'
                flag = 1
            if flag == 1: break

print(answer)

File: Softeer/Lv3.스마트_물류.cpp
#include<iostream>
#include<string>

using namespace std;

int main(int argc, char** argv)
{
	int n, k;
	cin >> n >> k;

	string factory;
	cin >> factory;

	int answer = 0;
	for(int i=0; i<factory.size(); i++) {
		if (factory[i] == 'P') {
			for(int j=i-k; j<i+k+1; j++) {
				if (j<0 || j==i || j>=factory.size()) continue;
				if (factory[j] == 'H'){
					answer += 1;
					factory[j] = 'N';
					break;
				}
			}
		}
	}

	cout << answer << endl;
	return 0;
}
